tests/dummy/app/services/gwr-config.js

```diff
@@ -0,0 +1,29 @@
+import Service from "@ember/service";
+
+export default class GwrConfigService extends Service {
+  get cssClasses() {
```

Comment:
What is this all about?

Reply:
The engine has to be configurable in regards to styling since we may have to integrate it in non uikit environments. so all layout styling is done with uikit and the all the other classes can be configured here to adapt to the host apps styling (f.i. bootstrap).

Reply:
Does this app only use UIkit for styling? So. No `uk-grid`, `uk-icon`, `uk-...`?

If you intend to use UIkit features that use anything other than CSS, you cannot just drop the thing into a non-UIkit application. At least not without including UIkit anyway and thus having two libraries bundled and initialized.

Reply:
If you have a look at the scss file, we only import the scss layouting stuff which does not require js.

Reply:
You [installed `ember-uikit`](https://github.com/adfinis-sygroup/ember-ebau-gwr/pull/1/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R35). Correct me if I'm wrong. But as far as I know this initialized everything. Regardless of what you import in your SCSS file.

Plus. The markup of those libraries might not match.

I don't want to say that it is impossible to have an addon work with multiple UI libraries but this is something that has to be thought through to the end.

To give you some inputs from existing addons:
* [ember-notify](https://github.com/adopted-ember-addons/ember-notify#turn-off-loading-css) has an option to disable importing of the addon's CSS in order to provide your own. The downside here is that the Javascript has to be provided/implemented by the addon itself (which is the case for ember-notify). But this would not allow for a drop-in experience as the developer would have to add the classes to his own style sheets and @extend the appropriate classes form his UI library.
* [ember-validated-form](https://github.com/adfinis-sygroup/ember-validated-form/blob/master/addon/-private/themed-component.js) introduced the idea of themed components so that the markup can be tailor-made to the different libraries.

We can also try it your way. But I highly recommend you sleep on it.

Reply:
Also. @czosel once said that we are creating UIkit applications with only minor CSS tweaks to approach the customer's corporate design. If we go this route the whole discussion becomes obsolete.

Reply:
ember-uikit has a env config to not import the js. this isnt turned on yet since this is still a prototype variant. We do not use uikit opinionated html structures. Uikit is only used for layouting via flex etc. And no `ember-uikit` does not auto import all of the uikit styling.

Reply:
> this isnt turned on yet

It isn't turned off and the default is `true`.

> And no ember-uikit does not auto import all of the uikit styling.

My comment about initializing everything was concerning the following statement:

> If you have a look at the scss file, we only import the scss layouting stuff which does not require js.

The opinion regarding the markup comes from the CSS selectors. Depending on what UIkit classes you use, you need to have the correct markup or otherwise the selectors won't work.

You list form elements in your class list. But if those elements don't just float somewhere on the page you will end up using other classes too (e.g. `uk-form-label`, `uk-form-controls`, `uk-inline`, ...). That's why I don't really see this list as final and thus try to think where this is heading.

I don't know what the final project will look like. Maybe this is all such a minor part that the whole discussion is overkill. But having CSS classes in the Javascript Code is a red flag to me and your explanation didn't really help alleviate my fears.

Reply:
@velrest I didn't follow all of the above, but I'm wondering if we even still need this to work without a full UIkit environment since the change in stakeholders?

Reply:
And maybe @anehx or @kaldras have some input. They haven't yet approve this PR either.

Reply:
@fkm-adfinis you can have a look at the form-prototype branch to see the html. It is not uikit opinionated as there are no opinionated classes used which would require a uikit specific layout. The forms are styled with flex mostly and width. no `uk-form-label` or `uk-form-controls`.

Reply:
> @velrest I didn't follow all of the above, but I'm wondering if we even still need this to work without a full UIkit environment since the change in stakeholders?

Yes that is a valid question. I will follow up on this privately so we can have a look at that.

Reply:
I've had a look at the other PR and there are a few issues with the current placement of classes. This might just be because it is a draft but this doesn't help me sleep :wink: But, let's wait for the outcome of your follow-up.